### PR TITLE
remove additional new lines in BakeFigure

### PR DESCRIPTION
**Pull request by ajluk**
For: https://github.com/openstax/cnx-recipes/issues/4786

### CODE REVIEW

**Comment by ajluk** [comment]
These changes solves the issue with white spaces, but caused failing `fizyka(pl-u-phyics recipe)` specs because of making one of the exercises lines display inline and wrap: 
image
image
I checked and the exercises looks good though in the content :thinking: :
image
image

File: lib/kitchen/directions/bake_figure.rb [hunk 1]
```diff
@@ -11,16 +11,16 @@ def self.v1(figure:, number:, cases: false)
         figure.target_label(label_text: 'figure', custom_content: number, cases: cases)
 
         title = figure.title&.cut
-        caption = figure.caption&.cut
+        caption = figure.caption&.wrap_children('span', class: 'os-caption')&.cut
```
Comment (ajluk) [inline lib/kitchen/directions/bake_figure.rb:14]: It is a workaround for the above below usage of nokogori `children` method in caption baking (l.23). If the actual children are wrapped inside another element in this case `span.os-caption`, the `children` method during parsing caption children doesn't add additional new lines between the nested children elements.

File: spec/recipes_spec/books/pl-u-physics/expected_output.xhtml [hunk 2]
```diff
@@ -33320,419 +33320,9 @@ podstaw podane w tre&#x15B;ci zadania warto&#x15B;ci drogi i czasu:<span data-ty
...
+                                                                                                                                                                                                            
+                                                                                                                                                                                                          </div>
```
Comment (ajluk) [inline spec/recipes_spec/books/pl-u-physics/expected_output.xhtml:33325]: Here it is visible the change in pl-u-physics test content. I have updated the `expected_output file` within the `actual_outout`.

File: lib/kitchen/directions/bake_figure.rb [hunk 3]
```diff
@@ -11,16 +11,21 @@ def self.v1(figure:, number:, cases: false)
         figure.target_label(label_text: 'figure', custom_content: number, cases: cases)
 
         title = figure.title&.cut
+        # caption = figure.caption&.wrap_children('span', class: 'os-caption')&.cut
```
Comment (kswanson33) [inline lib/kitchen/directions/bake_figure.rb:14]: don't forget to remove comment

**Review by kswanson33 — APPROVED** [review]
looking good! one note about the stray comment.

The exercise thing is baffling, but not bad. [Nokogiri does do some stuff to trim newlines](https://nokogiri.org/rdoc/Nokogiri/HTML5.html#module-Nokogiri::HTML5-label-HTML+Serialization), but no idea why changing whitespace in figures would make exercises elsewhere in the text get trimmed.

**Pull request merged**